nomad/upload_metrics.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+# this script is executed on Nomad clients, after a scenario scenario has been run, to upload metrics with telegraf.
+# It also takes care of setting the RUN_ID if unset from env
+
+set -euo pipefail
+
+function check_envset() {
+  local var_name="$1"
+  # temporarily allow fail to bind
+  set +u
+  if [[ -z "${!var_name}" ]]; then
+    echo "Environment variable $var_name is not set." >&2
+    set -u
+    exit 1
+  fi
+  set -u
+}
+
+check_envset "WT_METRICS_DIR"
+check_envset "INFLUX_TOKEN"
+check_envset "TELEGRAF_CONFIG_PATH"
+
+# read RUN_ID if not set in the environment
+if [ -z "${RUN_ID-}" ]; then
```

Comment:
Couldn't this also use `-v`?
```suggestion
if [ -v "$RUN_ID" ]; then
```